### PR TITLE
Migrate camel cased flags to kebab case + enable backward compatibility

### PR DESCRIPTION
**Pull request by khareyash05**
#### Describe the changes you've made
Migrate camel cased flags to kebab case
handle internal migration from camel case to kebab case
Needs doc changes

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Code style update (formatting, local variables)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Please let us know if any test cases are added

Please describe the tests(if any). Provide instructions how its affecting the coverage.

#### Describe if there is any unusual behaviour of your code(Write `NA` if there isn't)
NA

### CODE REVIEW

**Comment by khareyash05** [comment]
The output yml file is not converting to kebab case, working on the same

File: .github/workflows/test_workflow_scripts/golang-docker.sh [hunk 1]
```diff
@@ -64,7 +64,7 @@ send_request(){
 for i in {1..2}; do
     container_name="ginApp_${i}"
     send_request &
-    sudo -E env PATH=$PATH ./../../keployv2 record -c "docker run -p8080:8080 --net keploy-network --rm --name $container_name gin-mongo" --containerName "$container_name" --generateGithubActions=false &> "${container_name}.txt"
+    sudo -E env PATH=$PATH ./../../keployv2 record -c "docker run -p8080:8080 --net keploy-network --rm --name $container_name gin-mongo" --container-name "$container_name" --generate-github-actions=false &> "${container_name}.txt"
```
Comment (charankamarapu) [inline .github/workflows/test_workflow_scripts/golang-docker.sh:67]: use both camelCase and kebab type of flags in the workflow actions.

File: config/config.go [hunk 2]
```diff
@@ -8,55 +8,55 @@ import (
 )
 
 type Config struct {
-	Path                  string       `json:"path" yaml:"path" mapstructure:"path" `
-	AppID                 string       `json:"appId" yaml:"appId" mapstructure:"appId"`
+	Path                  string       `json:"path" yaml:"path" mapstructure:"path"`
```
Comment (charankamarapu) [inline config/config.go:11]: See if this change is not needed because you are binding camelCase flags to viper

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: config/config.go [hunk 3]
```diff
@@ -90,7 +90,7 @@ type Test struct {
 
 type Globalnoise struct {
 	Global   GlobalNoise  `json:"global" yaml:"global" mapstructure:"global"`
-	Testsets TestsetNoise `json:"test-sets" yaml:"test-sets" mapstructure:"test-sets"`
+	Testsets TestsetNoise `json:"testSets" yaml:"testSets" mapstructure:"test-sets"`
```
Comment (charankamarapu) [inline config/config.go:93]: Is this change required..?

**Review by charankamarapu — COMMENTED** [review]
Please address the comments

File: config/config.go (hunk 3 above)
Comment (khareyash05) [reply config/config.go]: missed that! on it

**Review by charankamarapu — APPROVED** [review]
LGTM
